src/stores/store/secrets/hashicorp/hashicorp.go

```diff
@@ -135,23 +135,79 @@ func (s *Store) List(_ context.Context) ([]string, error) {
 }
 
 func (s *Store) Delete(_ context.Context, id string) error {
-	return errors.ErrNotImplemented
+	logger := s.logger.With("id", id)
+
+	var callData map[string][]string
+	hashicorpSecretData, err := s.client.Read(s.pathData(id), callData)
+	if err != nil {
+		errMessage := "failed to get Hashicorp secret data for deletion"
+		logger.WithError(err).Error(errMessage)
+		return errors.FromError(err).SetMessage(errMessage)
+	}
+	if hashicorpSecretData == nil {
+		errMessage := "Hashicorp secret not found for deletion"
+		logger.WithError(err).Error(errMessage)
+		return errors.NotFoundError(errMessage)
+	}
+
+	err = s.client.Delete(s.pathData(id))
+	if err != nil {
+		errMessage := "failed to delete Hashicorp secret"
+		logger.WithError(err).Error(errMessage)
+		return errors.FromError(err).SetMessage(errMessage)
+	}
+
+	return nil
 }
 
 func (s *Store) GetDeleted(_ context.Context, id string) (*entities.Secret, error) {
 	return nil, errors.ErrNotImplemented
```

Comment:
not implemented or not supported?

Reply:
Not implemented since we will be able to do that once we have a DB

Reply:
then it's not supported on the Hashicorp side (store) but supported in the QKM (connector). Here it should be not supported as it cannot be done in Hashicorp